src/selfmain.c

```diff
@@ -79,31 +90,43 @@ void setBootProt(int v) {
         return;
     }
 
-    fuses[0] = (fuses[0] & ~NVMCTRL_FUSES_BOOTPROT_Msk) | (v << NVMCTRL_FUSES_BOOTPROT_Pos);
+    newfuses[0] = (newfuses[0] & ~NVMCTRL_FUSES_BOOTPROT_Msk) | (v << NVMCTRL_FUSES_BOOTPROT_Pos);
 
-    #ifdef SAMD21
-    NVMCTRL->CTRLB.reg = NVMCTRL->CTRLB.reg | NVMCTRL_CTRLB_CACHEDIS | NVMCTRL_CTRLB_MANW;
+    bool format = false;
+    for (int i = 0; i < sizeof(newfuses); ++i)
```

Comment:
`sizeof(newfuses)` is size in bytes. You want `sizeof(newfuses) / sizeof(newfuses[0])`

Reply:
fixed